app/models/location.rb

```diff
@@ -370,7 +370,10 @@ def postcode=(postcode)
   end
 
   def available_from
-    startdate || [created_at, FormHandler.instance.current_collection_start_date].min
+    return startdate if startdate.present?
+
+    window_end_date = Time.zone.local(created_at.year, 4, 1)
+    created_at < window_end_date ? Time.zone.local(created_at.year - 1, 4, 1) : Time.zone.local(created_at.year, 4, 1)
```

Comment:
Too many magic dates! We should take this from the form. Imagine we need to change it later...

Reply:
Hm, yeah, okay. Not looking at a specific form, because we wouldn't have one at this point, but moved it to the form handler. How's that?

Reply:
How do you know that April 1st is always going to be the date a form starts?

Reply:
Hm, has it changed in the past?
It's used across the codebase, both in form handler and in form model. I don't think there's any other date I could get at the moment. Did you have a different approach in mind?